File: test_medconnect.js
/**
 * MedConnect – Full System Selenium Test Suite (JavaScript)
 * ==========================================================
 * TC-01  Admin Login  – valid credentials
 * TC-02  Admin Login  – invalid credentials (error message shown)
 * TC-03  Admin Dashboard loads (sidebar visible)
 * TC-04  Admin Medical Centers section loads
 * TC-05  Admin Users & Staff – two tables present
 * TC-06  Admin Pending Staff section in DOM
 * TC-07  Staff Registration form loads & empty submit blocked
 * TC-08  Patient Login  – invalid credentials (error shown)
 * TC-09  Patient Registration form toggle works
 * TC-10  Staff Login   – invalid credentials (error shown)
 * TC-11  Medical Centers API returns active centers
 * TC-12  Landing / Index page loads
 *
 * Requirements (already installed):
 *   npm install selenium-webdriver chromedriver
 *
 * Run:
 *   node test_medconnect.js
 */

const { Builder, By, until } = require("selenium-webdriver");
const chrome = require("selenium-webdriver/chrome");
const http   = require("http");
require("chromedriver");

// ─── CONFIG ─────────────────────────────────────────────────────────────────
const BASE_URL       = "http://localhost:8000";
const ADMIN_EMAIL    = "admin@medconnect.com";  // ← change if different
const ADMIN_PASSWORD = "admin123";          // ← change if different
const TIMEOUT        = 10_000;             // ms
const HEADLESS       = false;              // true = no browser window
// ─────────────────────────────────────────────────────────────────────────────

// ─── RESULT TRACKING ─────────────────────────────────────────────────────────
let passed = 0, failed = 0;
const results = [];

function pass(name) {
  passed++;
  results.push({ name, status: "PASS" });
  console.log(`  ✅ PASS – ${name}`);
}

function fail(name, err) {
  failed++;
  results.push({ name, status: "FAIL", error: err.message });
  console.log(`  ❌ FAIL – ${name}`);
  console.log(`       ${err.message}`);
}
// ─────────────────────────────────────────────────────────────────────────────

async function buildDriver() {
  const opts = new chrome.Options();
  if (HEADLESS) opts.addArguments("--headless=new");
  opts.addArguments("--window-size=1400,900", "--no-sandbox", "--disable-dev-shm-usage", "--log-level=3");
  return new Builder().forBrowser("chrome").setChromeOptions(opts).build();
}

async function go(driver, path = "") {
  await driver.get(`${BASE_URL}/${path.replace(/^\//, "")}`);
  await driver.sleep(600);
}

async function adminLogin(driver) {
  await go(driver, "admin_login.html");
  await driver.wait(until.elementLocated(By.id("email")), TIMEOUT);
  await driver.findElement(By.id("email")).sendKeys(ADMIN_EMAIL);
  await driver.findElement(By.id("password")).sendKeys(ADMIN_PASSWORD);
  await driver.findElement(By.id("admin-submit-btn")).click();
  await driver.wait(until.urlContains("dashboard_admin"), TIMEOUT);
}

async function getAlertText(driver, id) {
  try {
    return await driver.findElement(By.id(id)).getText();
  } catch { return ""; }
}

// ─────────────────────────────────────────────────────────────────────────────
// TEST RUNNER
// ─────────────────────────────────────────────────────────────────────────────
async function runTests() {
  console.log("=".repeat(60));
  console.log("  MedConnect – Full Selenium Test Suite (JavaScript)");
  console.log(`  Server : ${BASE_URL}`);
  console.log("=".repeat(60));

  const driver = await buildDriver();

  try {
    // ── TC-01  Admin Login – valid ──────────────────────────────────────────
    console.log("\n[TC-01] Admin Login – valid credentials");
    try {
      await driver.manage().deleteAllCookies();
      await go(driver, "admin_login.html");
      await driver.wait(until.elementLocated(By.id("email")), TIMEOUT);
      await driver.findElement(By.id("email")).sendKeys(ADMIN_EMAIL);
      await driver.findElement(By.id("password")).sendKeys(ADMIN_PASSWORD);
      await driver.findElement(By.id("admin-submit-btn")).click();
      await driver.wait(until.urlContains("dashboard_admin"), TIMEOUT);
      const url = await driver.getCurrentUrl();
      if (!url.includes("dashboard_admin")) throw new Error("Not redirected to dashboard");
      pass("Admin Login – valid credentials");
    } catch (e) { fail("Admin Login – valid credentials", e); }

    // ── TC-02  Admin Login – invalid ────────────────────────────────────────
    console.log("\n[TC-02] Admin Login – invalid credentials");
    try {
      await driver.manage().deleteAllCookies();
      await go(driver, "admin_login.html");
      await driver.wait(until.elementLocated(By.id("email")), TIMEOUT);
      await driver.findElement(By.id("email")).sendKeys("wrong@test.com");
      await driver.findElement(By.id("password")).sendKeys("badpassword");
      await driver.findElement(By.id("admin-submit-btn")).click();
      await driver.sleep(2000);
      const alertText = await getAlertText(driver, "admin-alert-box");
      if (!alertText) throw new Error("No error message displayed");
      const url = await driver.getCurrentUrl();
      if (url.includes("dashboard_admin")) throw new Error("Should not redirect with wrong credentials");
      pass("Admin Login – invalid credentials");
    } catch (e) { fail("Admin Login – invalid credentials", e); }

    // ── TC-03  Admin Dashboard loads ────────────────────────────────────────
    console.log("\n[TC-03] Admin Dashboard loads");
    try {
      await driver.manage().deleteAllCookies();
      await adminLogin(driver);
      const title = await driver.getTitle();
      if (!title.toLowerCase().includes("admin")) throw new Error(`Unexpected title: ${title}`);
      await driver.findElement(By.className("sidebar")); // throws if not found
      pass("Admin Dashboard loads");
    } catch (e) { fail("Admin Dashboard loads", e); }

    // ── TC-04  Admin Medical Centers section ────────────────────────────────
    console.log("\n[TC-04] Admin Medical Centers section");
    try {
      await driver.manage().deleteAllCookies();
      await adminLogin(driver);
      // Click Centers nav link
      const links = await driver.findElements(By.css(".sidebar a, .nav-item"));
      for (const link of links) {
        const txt = (await link.getText()).toLowerCase();
        if (txt.includes("center")) { await link.click(); break; }
      }
      await driver.sleep(1500);
      const cards = await driver.findElements(By.className("card"));
      if (cards.length === 0) throw new Error("No cards found in Centers section");
      pass(`Admin Medical Centers section (${cards.length} card(s))`);
    } catch (e) { fail("Admin Medical Centers section", e); }

    // ── TC-05  Admin Users & Staff – two tables ──────────────────────────────
    console.log("\n[TC-05] Admin Users & Staff – Patients and Staff tables");
    try {
      await driver.manage().deleteAllCookies();
      await adminLogin(driver);
      const links = await driver.findElements(By.css(".sidebar a, .nav-item"));
      for (const link of links) {
        const txt = (await link.getText()).toLowerCase();
        if (txt.includes("user")) { await link.click(); break; }
      }
      await driver.sleep(1500);
      const tables = await driver.findElements(By.tagName("table"));
      if (tables.length < 2) throw new Error(`Expected ≥2 tables, found ${tables.length}`);
      pass(`Admin Users & Staff (${tables.length} table(s) found)`);
    } catch (e) { fail("Admin Users & Staff tables", e); }

    // ── TC-06  Admin Pending Staff section in DOM ────────────────────────────
    console.log("\n[TC-06] Admin Pending Staff section in DOM");
    try {
      await driver.manage().deleteAllCookies();
      await adminLogin(driver);
      const el = await driver.findElement(By.id("pendingStaffCard"));
      if (!el) throw new Error("pendingStaffCard not in DOM");
      pass("Admin Pending Staff section in DOM");
    } catch (e) { fail("Admin Pending Staff section in DOM", e); }

    // ── TC-07  Staff Registration form ──────────────────────────────────────
    console.log("\n[TC-07] Staff Registration form loads");
    try {
      await driver.manage().deleteAllCookies();
      await go(driver, "register_staff.html");
      await driver.wait(until.elementLocated(By.id("staffRegisterForm")), TIMEOUT);
      // Submit empty form – browser validation should block it
      await driver.findElement(By.css("#staffRegisterForm button[type='submit']")).click();
      await driver.sleep(800);
      const url = await driver.getCurrentUrl();
      if (!url.includes("register_staff")) throw new Error("Form submitted without validation!");
      pass("Staff Registration form loads & validation works");
    } catch (e) { fail("Staff Registration form", e); }

    // ── TC-08  Patient Login – invalid ──────────────────────────────────────
    console.log("\n[TC-08] Patient Login – invalid credentials");
    try {
      await driver.manage().deleteAllCookies();
      await go(driver, "login.html");
      await driver.wait(until.elementLocated(By.css("input[name='first_name']")), TIMEOUT);
      await driver.findElement(By.css("input[name='first_name']")).sendKeys("FakeUser");
      await driver.findElement(By.css("input[name='id_number']")).sendKeys("000000000000");
      await driver.findElement(By.id("patient-submit-btn")).click();
      await driver.sleep(2000);
      const alertText = await getAlertText(driver, "patient-alert-box");
      if (!alertText) throw new Error("No error message displayed for invalid patient login");
      pass(`Patient Login – invalid (msg: "${alertText}")`);
    } catch (e) { fail("Patient Login – invalid credentials", e); }

    // ── TC-09  Patient Registration toggle ──────────────────────────────────
    console.log("\n[TC-09] Patient Registration form toggle");
    try {
      await driver.manage().deleteAllCookies();
      await go(driver, "login.html");
      await driver.wait(until.elementLocated(By.id("patientLoginForm")), TIMEOUT);
      const regLink = await driver.findElement(By.xpath("//a[contains(text(),'Register here')]"));
      await regLink.click();
      await driver.sleep(500);
      const regForm = await driver.findElement(By.id("registerForm"));
      const visible = await regForm.isDisplayed();
      if (!visible) throw new Error("Registration form not visible after toggle");
      pass("Patient Registration form toggle works");
    } catch (e) { fail("Patient Registration form toggle", e); }

    // ── TC-10  Staff Login – invalid ─────────────────────────────────────────
    console.log("\n[TC-10] Staff Login – invalid credentials");
    try {
      await driver.manage().deleteAllCookies();
      await go(driver, "staff_login.html");
      await driver.wait(until.elementLocated(By.id("username")), TIMEOUT);
      await driver.findElement(By.id("username")).sendKeys("fakestaff@clinic.com");
      await driver.findElement(By.id("password")).sendKeys("wrongpassword");
      await driver.findElement(By.id("staff-submit-btn")).click();
      await driver.sleep(2000);
      const alertText = await getAlertText(driver, "staff-alert-box");
      if (!alertText) throw new Error("No error message for invalid staff login");
      pass(`Staff Login – invalid (msg: "${alertText}")`);
    } catch (e) { fail("Staff Login – invalid credentials", e); }

    // ── TC-11  Medical Centers API ───────────────────────────────────────────
    console.log("\n[TC-11] Medical Centers API returns data");
    try {
      await new Promise((resolve, reject) => {
        http.get(`${BASE_URL}/api/get_centers.php`, (res) => {
          let data = "";
          res.on("data", chunk => data += chunk);
          res.on("end", () => {
            try {
              const json = JSON.parse(data);
              if (json.status !== "success") throw new Error(`status = ${json.status}`);
              if (!Array.isArray(json.data))  throw new Error("data is not an array");
              console.log(`       API returned ${json.data.length} active center(s)`);
              resolve();
            } catch (err) { reject(err); }
          });
        }).on("error", reject);
      });
      pass("Medical Centers API returns active centers");
    } catch (e) { fail("Medical Centers API", e); }

    // ── TC-12  Landing / Index page ──────────────────────────────────────────
    console.log("\n[TC-12] Landing / Index page loads");
    try {
      await driver.manage().deleteAllCookies();
      await go(driver, "index.html");
      await driver.wait(until.elementLocated(By.tagName("body")), TIMEOUT);
      const title = await driver.getTitle();
      if (!title) throw new Error("Page title is empty");
      pass(`Index page loads (title: "${title}")`);
    } catch (e) { fail("Index page loads", e); }

  } finally {
    await driver.quit();
  }

  // ── SUMMARY ────────────────────────────────────────────────────────────────
  console.log("\n" + "=".repeat(60));
  console.log("  TEST RESULTS SUMMARY");
  console.log("=".repeat(60));
  results.forEach((r, i) => {
    const icon = r.status === "PASS" ? "✅" : "❌";
    console.log(`  ${String(i + 1).padStart(2, "0")}. ${icon} ${r.status}  –  ${r.name}`);
    if (r.error) console.log(`        Error: ${r.error}`);
  });
  console.log("─".repeat(60));
  console.log(`  Total: ${results.length}  |  Passed: ${passed}  |  Failed: ${failed}`);
  console.log("=".repeat(60));

  if (failed > 0) process.exit(1);
}

runTests().catch(err => {
  console.error("\nFATAL:", err.message);
  process.exit(1);
});

File: test_medconnect.py
"""
MedConnect - Full System Selenium Test Suite
=============================================
Covers all major user flows:
  TC-01  Admin Login  (valid)
  TC-02  Admin Login  (invalid credentials)
  TC-03  Admin Dashboard loads & sidebar works
  TC-04  Admin Medical Centers page loads
  TC-05  Admin Users & Staff page loads
  TC-06  Admin Pending Staff Approvals visible
  TC-07  Staff Registration form loads & validation
  TC-08  Patient Login  (invalid – wrong NIC)
  TC-09  Patient Registration form visible on toggle
  TC-10  Staff Login  (invalid credentials)
  TC-11  Medical Centers API returns data
  TC-12  Index / Landing page loads

Requirements:
  pip install selenium webdriver-manager

Run:
  py test_medconnect.py
"""

import time
import unittest

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from webdriver_manager.chrome import ChromeDriverManager

# ─── CONFIG ────────────────────────────────────────────────────────────────────
BASE_URL        = "http://localhost:8000"
ADMIN_EMAIL     = "admin@medconnect"   # ← update if different
ADMIN_PASSWORD  = "admin123"           # ← update if different
WAIT_TIMEOUT    = 10                   # seconds
HEADLESS        = False                # set True to run without browser window
# ───────────────────────────────────────────────────────────────────────────────


def get_driver():
    options = Options()
    if HEADLESS:
        options.add_argument("--headless=new")
    options.add_argument("--window-size=1400,900")
    options.add_argument("--no-sandbox")
    options.add_argument("--disable-dev-shm-usage")
    options.add_experimental_option("excludeSwitches", ["enable-logging"])
    service = Service(ChromeDriverManager().install())
    return webdriver.Chrome(service=service, options=options)


class MedConnectTests(unittest.TestCase):
    """Complete Selenium test suite for MedConnect system."""

    @classmethod
    def setUpClass(cls):
        cls.driver = get_driver()
        cls.wait   = WebDriverWait(cls.driver, WAIT_TIMEOUT)
        cls.driver.implicitly_wait(5)

    @classmethod
    def tearDownClass(cls):
        cls.driver.quit()

    def setUp(self):
        """Clear cookies before each test so sessions don't leak."""
        self.driver.delete_all_cookies()

    # ─────────────────────────────────────────────────────────────────
    # HELPERS
    # ─────────────────────────────────────────────────────────────────

    def go(self, path=""):
        self.driver.get(BASE_URL + "/" + path.lstrip("/"))
        time.sleep(0.5)

    def get_alert_text(self, alert_id):
        try:
            el = self.driver.find_element(By.ID, alert_id)
            return el.text.strip()
        except Exception:
            return ""

    def admin_login(self):
        """Helper: perform a successful admin login."""
        self.go("admin_login.html")
        self.driver.find_element(By.ID, "email").send_keys(ADMIN_EMAIL)
        self.driver.find_element(By.ID, "password").send_keys(ADMIN_PASSWORD)
        self.driver.find_element(By.ID, "admin-submit-btn").click()
        # wait for redirect to dashboard
        self.wait.until(EC.url_contains("dashboard_admin"))

    # ─────────────────────────────────────────────────────────────────
    # TC-01  Admin Login – valid credentials
    # ─────────────────────────────────────────────────────────────────
    def test_01_admin_login_valid(self):
        """Admin should be redirected to dashboard after correct login."""
        print("\n[TC-01] Admin Login – valid credentials")
        self.go("admin_login.html")

        self.wait.until(EC.presence_of_element_located((By.ID, "email")))
        self.driver.find_element(By.ID, "email").clear()
        self.driver.find_element(By.ID, "email").send_keys(ADMIN_EMAIL)
        self.driver.find_element(By.ID, "password").clear()
        self.driver.find_element(By.ID, "password").send_keys(ADMIN_PASSWORD)
        self.driver.find_element(By.ID, "admin-submit-btn").click()

        self.wait.until(EC.url_contains("dashboard_admin"))
        self.assertIn("dashboard_admin", self.driver.current_url,
                      "Admin was NOT redirected to the dashboard!")
        print("    ✅  PASS – redirected to", self.driver.current_url)

    # ─────────────────────────────────────────────────────────────────
    # TC-02  Admin Login – invalid credentials
    # ─────────────────────────────────────────────────────────────────
    def test_02_admin_login_invalid(self):
        """Wrong password should show an error message."""
        print("\n[TC-02] Admin Login – invalid credentials")
        self.go("admin_login.html")

        self.wait.until(EC.presence_of_element_located((By.ID, "email")))
        self.driver.find_element(By.ID, "email").send_keys("wrong@admin.com")
        self.driver.find_element(By.ID, "password").send_keys("wrongpassword")
        self.driver.find_element(By.ID, "admin-submit-btn").click()

        time.sleep(2)
        alert = self.get_alert_text("admin-alert-box")
        self.assertTrue(len(alert) > 0,
                        "Expected an error message but got nothing!")
        self.assertNotIn("dashboard_admin", self.driver.current_url,
                         "Should NOT have redirected to dashboard with wrong credentials!")
        print("    ✅  PASS – error shown:", alert)

    # ─────────────────────────────────────────────────────────────────
    # TC-03  Admin Dashboard – page loads & sidebar navigation
    # ─────────────────────────────────────────────────────────────────
    def test_03_admin_dashboard_loads(self):
        """Dashboard should load and sidebar links should be present."""
        print("\n[TC-03] Admin Dashboard – page loads")
        self.admin_login()

        title = self.driver.title
        self.assertIn("Admin", title, f"Dashboard page title unexpected: '{title}'")

        # Check sidebar exists
        sidebar = self.driver.find_element(By.CLASS_NAME, "sidebar")
        self.assertIsNotNone(sidebar, "Sidebar not found on dashboard!")
        print("    ✅  PASS – Dashboard loaded, title:", title)

    # ─────────────────────────────────────────────────────────────────
    # TC-04  Admin Dashboard – Medical Centers section
    # ─────────────────────────────────────────────────────────────────
    def test_04_admin_centers_section(self):
        """Medical Centers section should load data."""
        print("\n[TC-04] Admin Dashboard – Medical Centers section")
        self.admin_login()

        # Click on Medical Centers nav item
        try:
            centers_link = self.driver.find_element(
                By.XPATH, "//a[contains(@onclick, 'centers') or contains(text(), 'Center')]"
            )
            centers_link.click()
        except Exception:
            # Try sidebar nav links
            nav_links = self.driver.find_elements(By.CSS_SELECTOR, ".sidebar a, .nav-item")
            for link in nav_links:
                if "center" in link.text.lower():
                    link.click()
                    break

        time.sleep(2)
        # Verify center cards / table are visible
        cards = self.driver.find_elements(By.CLASS_NAME, "card")
        self.assertGreater(len(cards), 0, "No cards found in Medical Centers section!")
        print(f"    ✅  PASS – Found {len(cards)} card(s) in Centers section")

    # ─────────────────────────────────────────────────────────────────
    # TC-05  Admin Dashboard – Users & Staff section
    # ─────────────────────────────────────────────────────────────────
    def test_05_admin_users_section(self):
        """Users & Staff section should have Patients and Staff tables."""
        print("\n[TC-05] Admin Dashboard – Users & Staff section")
        self.admin_login()

        try:
            users_link = self.driver.find_element(
                By.XPATH, "//a[contains(text(),'User') or contains(@onclick,'users')]"
            )
            users_link.click()
        except Exception:
            nav_links = self.driver.find_elements(By.CSS_SELECTOR, ".sidebar a, .nav-item")
            for link in nav_links:
                if "user" in link.text.lower():
                    link.click()
                    break

        time.sleep(2)
        # Patients and Staff tables should exist
        tables = self.driver.find_elements(By.TAG_NAME, "table")
        self.assertGreaterEqual(len(tables), 2,
                                f"Expected at least 2 tables (Patients + Staff), got {len(tables)}")
        print(f"    ✅  PASS – Found {len(tables)} table(s) in Users section")

    # ─────────────────────────────────────────────────────────────────
    # TC-06  Admin Dashboard – Pending Staff section visible
    # ─────────────────────────────────────────────────────────────────
    def test_06_admin_pending_staff_section(self):
        """Pending Staff Approvals section should be present in the DOM."""
        print("\n[TC-06] Admin Dashboard – Pending Staff section in DOM")
        self.admin_login()

        pending_card = self.driver.find_element(By.ID, "pendingStaffCard")
        self.assertIsNotNone(pending_card, "pendingStaffCard element not found!")
        print("    ✅  PASS – pendingStaffCard element is in the DOM")

    # ─────────────────────────────────────────────────────────────────
    # TC-07  Staff Registration – form loads & required fields
    # ─────────────────────────────────────────────────────────────────
    def test_07_staff_registration_form(self):
        """Staff registration form should load and require fields."""
        print("\n[TC-07] Staff Registration – form loads")
        self.go("register_staff.html")

        self.wait.until(EC.presence_of_element_located((By.ID, "staffRegisterForm")))
        form = self.driver.find_element(By.ID, "staffRegisterForm")
        self.assertIsNotNone(form, "Staff registration form not found!")

        # Try submitting empty form and check it doesn't proceed
        self.driver.find_element(By.CSS_SELECTOR, "#staffRegisterForm button[type='submit']").click()
        time.sleep(1)
        # Should still be on register page (browser validation prevents submit)
        self.assertIn("register_staff", self.driver.current_url)
        print("    ✅  PASS – Form loads and validation prevents empty submission")

    # ─────────────────────────────────────────────────────────────────
    # TC-08  Patient Login – invalid credentials
    # ─────────────────────────────────────────────────────────────────
    def test_08_patient_login_invalid(self):
        """Patient login with wrong NIC should show an error."""
        print("\n[TC-08] Patient Login – invalid credentials")
        self.go("login.html")

        self.wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, "input[name='first_name']")))
        self.driver.find_element(By.CSS_SELECTOR, "input[name='first_name']").send_keys("FakeUser")
        self.driver.find_element(By.CSS_SELECTOR, "input[name='id_number']").send_keys("000000000000")
        self.driver.find_element(By.ID, "patient-submit-btn").click()

        time.sleep(2)
        alert = self.get_alert_text("patient-alert-box")
        self.assertTrue(len(alert) > 0, "Expected error message for invalid patient login!")
        print("    ✅  PASS – Error message shown:", alert)

    # ─────────────────────────────────────────────────────────────────
    # TC-09  Patient Registration – form toggle works
    # ─────────────────────────────────────────────────────────────────
    def test_09_patient_register_toggle(self):
        """'Register here' link should show the registration form."""
        print("\n[TC-09] Patient – registration form toggle")
        self.go("login.html")

        self.wait.until(EC.presence_of_element_located((By.ID, "patientLoginForm")))

        # Click Register link
        register_link = self.driver.find_element(
            By.XPATH, "//a[contains(text(),'Register here')]"
        )
        register_link.click()
        time.sleep(0.5)

        reg_form = self.driver.find_element(By.ID, "registerForm")
        is_visible = reg_form.is_displayed()
        self.assertTrue(is_visible, "Registration form should be visible after clicking 'Register here'!")
        print("    ✅  PASS – Registration form is now visible")

    # ─────────────────────────────────────────────────────────────────
    # TC-10  Staff Login – invalid credentials
    # ─────────────────────────────────────────────────────────────────
    def test_10_staff_login_invalid(self):
        """Staff login with wrong password should show an error."""
        print("\n[TC-10] Staff Login – invalid credentials")
        self.go("staff_login.html")

        self.wait.until(EC.presence_of_element_located((By.ID, "username")))
        self.driver.find_element(By.ID, "username").send_keys("fakestaff@clinic.com")
        self.driver.find_element(By.ID, "password").send_keys("wrongpassword")
        self.driver.find_element(By.ID, "staff-submit-btn").click()

        time.sleep(2)
        alert = self.get_alert_text("staff-alert-box")
        self.assertTrue(len(alert) > 0, "Expected error message for invalid staff login!")
        print("    ✅  PASS – Error message shown:", alert)

    # ─────────────────────────────────────────────────────────────────
    # TC-11  Medical Centers API – returns data
    # ─────────────────────────────────────────────────────────────────
    def test_11_get_centers_api(self):
        """The get_centers API should return a JSON success response."""
        print("\n[TC-11] Medical Centers API – returns data")
        import urllib.request, json
        url = BASE_URL + "/api/get_centers.php"
        with urllib.request.urlopen(url, timeout=10) as resp:
            data = json.loads(resp.read().decode())

        self.assertEqual(data.get("status"), "success",
                         f"API status not 'success': {data}")
        centers = data.get("data", [])
        self.assertIsInstance(centers, list, "'data' should be a list")
        print(f"    ✅  PASS – API returned {len(centers)} active center(s)")

    # ─────────────────────────────────────────────────────────────────
    # TC-12  Landing / Index page loads
    # ─────────────────────────────────────────────────────────────────
    def test_12_index_page_loads(self):
        """The main index page should load successfully."""
        print("\n[TC-12] Index/Landing page loads")
        self.go("index.html")

        self.wait.until(EC.presence_of_element_located((By.TAG_NAME, "body")))
        title = self.driver.title
        self.assertTrue(len(title) > 0, "Page title is empty!")
        body_text = self.driver.find_element(By.TAG_NAME, "body").text
        self.assertTrue(len(body_text) > 10, "Page body appears to be empty!")
        print(f"    ✅  PASS – Page loaded, title: '{title}'")


# ─── ENTRY POINT ───────────────────────────────────────────────────────────────
if __name__ == "__main__":
    print("=" * 60)
    print("  MedConnect – Full Selenium Test Suite")
    print(f"  Server : {BASE_URL}")
    print("=" * 60)
    unittest.main(verbosity=0, failfast=False)
